jmclient/jmclient/blockchaininterface.py

```diff
@@ -990,6 +992,6 @@ def get_received_by_addr(self, addresses, query_params):
         for address in addresses:
             #self.rpc('importaddress', [address, 'watchonly'])
             res.append({'address': address,
-                        'balance': int(round(Decimal(1e8) * Decimal(self.rpc(
```

Comment:
First to note that `get_received_by_addr` isn't really useful most of the time, and it's mostly unused since very early in the project (was only ever used for regtest tests, of course). Just one usage in test_segwit.

This `round` call was unhelpful and unneeded, but I know where it comes from.
If you do just:

```python
int(decimal.Decimal(1e8)*decimal.Decimal(1.23456789))
123456788
```

whereas if you do

```python
int(decimal.Decimal(1e8)*decimal.Decimal('1.23456789'))
123456789
```

In cases where you want to deal with numbers (not rpc call returns for coin amounts, which are Decimal objects), you need to either change the precision setting in the `decimal` module or else use nasty syntax like int(round(float(...)))

We are using it on string returns from the rpc, except `estimate_fee_per_kb` where it's only an estimate anyway.

Reply:
The RPC call should now return decimals directly so these shouldn't require rounding.